### PR TITLE
Fdp35 my posts

### PR DESCRIPTION
**Pull request by TomerMiz10**
## Task link from JIRA:
https://mssplinter10-1711638164183.atlassian.net/jira/software/projects/FDP/boards/1?selectedIssue=FDP-35

## Description:
- chore: Remove unnecessary interfaces and lines, delete console log statements. (9dab382)
- Refactor: `EditProfilePage` to remove userName field, Remove console log statements (2acad4b)
- feat: Add toggle button to show all/my posts in HomePage (b3670ad)
- Feat: Add `getPostsByUser` function {description} API endpoint. And toggle button to show all\my posts in `HomePage`. (b088044)
- Refactor: Remove unnecessary imports from `EditProfilePage.tsx` . (02c4048)
- Refactor `EditProfilePage` component to improve code readability and maintainability (ccc5ef8)
- chore: Refactor `EditProfilePage` component to improve code readability and maintainability (3707d73)
- Merge PR #22 from branch 'master' of https://github.com/nitzanto/Foodies-WebApp into `FDP35-my-posts` (c82043a)
- chore: Update `selectedPost` comments handling in `HomePage.tsx` (c2f67ed)
- feat: Add `Book`, `Comment`, `Post`, and `Review` interfaces to frontend. (7453265)
- feat: Add `_id` attribute to `Post` and `Review` interfaces for creating posts. (07f5dfc)

## Code Owners:
@TomerMiz10 

### Summary
Display MyPosts in HomePage.
Edit profile details refinements.

### CODE REVIEW

File: frontend/src/pages/ProfilePage/EditProfilePage.tsx
Comment (TomerMiz10) [inline frontend/src/pages/ProfilePage/EditProfilePage.tsx:1]: i fixed indentations here, and deleted `Grid` of `username`

File: backend/src/controllers/auth.controller.ts
Comment (nitzanto) [inline backend/src/controllers/auth.controller.ts:1]: Delete console logs

File: backend/src/interfaces/Post.ts
Comment (nitzanto) [inline backend/src/interfaces/Post.ts:1]: Why was this file re-created?

File: backend/src/interfaces/Review.ts
Comment (nitzanto) [inline backend/src/interfaces/Review.ts:1]: Why _id was added? this shouldnt be changed

**Review by nitzanto — CHANGES_REQUESTED** [review]

File: backend/src/controllers/auth.controller.ts
Comment (TomerMiz10) [reply backend/src/controllers/auth.controller.ts:1]: done

File: backend/src/interfaces/Review.ts
Comment (TomerMiz10) [reply backend/src/interfaces/Review.ts:1]: done

File: backend/src/interfaces/Review.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import mongoose from 'mongoose';
+
```
Comment (nitzanto) [inline backend/src/interfaces/Review.ts]: delete this mongoose from this file

**Review by nitzanto — CHANGES_REQUESTED** [review]

File: backend/src/interfaces/Post.ts
Comment (nitzanto) [inline backend/src/interfaces/Post.ts:1]: Why did you add this?

**Review by nitzanto — DISMISSED** [review]

**Review by omrifinegold111 — APPROVED** [review]

**Review by omrifinegold111 — APPROVED** [review]
nice

**Pull request merged**